### PR TITLE
Rename btrisolve to lu_solve

### PR DESCRIPTION
**Pull request by vishwakftw**
Changelog:
- Rename `btrisolve` to `lu_solve` to remain consistent with names of solve methods (`cholesky_solve`, `triangular_solve`, `solve`)
- Fix all callsites
- Rename all tests
- Create a tentative alias for `lu_solve` under the name `btrisolve` and add a deprecation warning to not promote usage

Test Plan:
- All tests should pass to confirm that the patch is correct

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
LGTM. The CI seems to not be working right now

**Comment by vishwakftw** [comment]
@zou3519, I have rebased and the CI is green (except for unrelated test failures). Is this good to go?

**Comment by zou3519** [comment]
image

I'm not sure if the lint is broken on master, but it looks like this might need another rebase?

**Comment by vishwakftw** [comment]
@zou3519 I have rebased it again; I didn’t notice the merge conflicts on my phone. Thank you for letting me know.

File: torch/functional.py [hunk 1]
```diff
@@ -821,6 +822,22 @@ def btriunpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
...
+
+    For more information regarding :func:`torch.btrisolve`, please check
+    :func:`torch.lu_solve`.
+
+    .. warning::
+        :func:`torch.btrisolve` is deprecated in favour of :func:`torch.lu_solve` and will be
+        removed in the next release. Please use :func:`torch.lu_solve` instead.
+    """
+    warnings.warn("torch.btrisolve is deprecated in favour of torch.lu_solve and will be "
+                  "removed in the next release. Please use torch.lu_solve instead.", stackleve=2)
```
Comment (zou3519) [inline torch/functional.py]: this needs to be changed to `stacklevel`

Comment (zou3519) [reply torch/functional.py]: (I made the change in-place)

Comment (vishwakftw) [reply torch/functional.py]: Oh, I'm sorry about that. Thank you for fixing it.

File: torch/functional.py [hunk 2]
```diff
@@ -821,6 +822,22 @@ def btriunpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
...
+    For more information regarding :func:`torch.btrisolve`, please check
+    :func:`torch.lu_solve`.
+
+    .. warning::
+        :func:`torch.btrisolve` is deprecated in favour of :func:`torch.lu_solve` and will be
+        removed in the next release. Please use :func:`torch.lu_solve` instead.
+    """
+    warnings.warn("torch.btrisolve is deprecated in favour of torch.lu_solve and will be "
+                  "removed in the next release. Please use torch.lu_solve instead.", stacklevel=2)
+    return torch.lu_solve(b=b, LU_data=LU_data, LU_pivots=LU_pivots, out=out)
```
Comment (zou3519) [inline torch/functional.py]: @vishwakftw I think `b=b` is wrong.  Calling this directly gets the following:

```
TypeError: lu_solve() missing 3 required positional argument: "input", "LU_data", "LU_pivots"
  File "/usr/local/fbcode/gcc-5-glibc-2.23/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
```

Comment (zou3519) [reply torch/functional.py]: (this should probably just be `torch.lu_solve(b, LU_data=LU_data, LU_pivots=LU_pivots, out=out)`). I don't know how native_functions.yaml names the first argument.

Comment (vishwakftw) [reply torch/functional.py]: I've changed it. Sorry about this.

Comment (zou3519) [reply torch/functional.py]: No worries, I missed these on my initial review

**Comment by vishwakftw** [comment]
@zou3519 is this good to go?

**Comment by zou3519** [comment]
It should be good to go, I've been struggling with landing it however

**Comment by vishwakftw** [comment]
Please let me know how I can help to get this landed.

**Pull request closed without merging**